docs/monoDrive_home/Simulator-Commands.md

```diff
@@ -0,0 +1,365 @@
+        "specular_exponent": (float) - the specular exponent of the material,
+        "specular_coefficient": (float) - the specular coefficient of the material,
+			"diffuse_coefficient": (float) - the diffuse coefficient of the material,
+			"dielectric_constant": (float) - the dielectric constant of the material,
+			"roughness": (float) - a value between 0.0 and 1.0 
+				corresponding to the roughness of the material
+        }
+	},
+	"traffic_configuration": {
+		"max_vehicles": (integer) - the maximum number of traffic vehicles at any given time 
+			during this simulation run,
+        "min_desired_speed": (float) - the minimum speed for traffic vehicles in cm/s,
+        "max_desired_speed": (float) - the maximum speed for traffic vehicles in cm/s,
+		"spawn_leash_distance": (float) - distance from ego vehicle at which traffic vehicles 
+			are removed from the scene,
+		"min_spawn_range": (float) - the minimum distance from the ego vehicle at which 
+			to spawn new traffic vehicles,
+		"swarm_traffic": (boolean) - whether to spawn traffic vehicles as 
+			one large group or individually,
+		"vertical_offset": (float) - an offset to apply to the Z direction when 
+			spawning traffic vehicles,
+		"overtake_slower_vehicles": (float) - whether to overtake or follow vehicles with a 
+			slower desired speed
+	}
+}
+```
+
+For more information on these values:
+
+ - [*specular exponent*](https://en.wikipedia.org/wiki/Specular_highlight)
+ - [*specular coefficient*](https://en.wikipedia.org/wiki/Specular_reflection)
+ - [*diffuse coefficient*](https://en.wikipedia.org/wiki/Mass_diffusivity)
+ - [*dielectric constant*](https://en.wikipedia.org/wiki/Relative_permittivity)
+
+**response**: (string)
+
+`simulator configuration mode = (0, 1, or 2)`  if the configure command was successful, or `invalid map specified` if an invalid map is requested
+
+
+## EGO Vehicle Control
+Configures the ego vehicle used in the simulation run.
+
+**command ID**:  "EgoControl_ID"
+
+**command data**:
+```json
+{
+	"forward_amount": (float) - value between 0.0 and 1.0 referring to the amount 
+of throttle to apply,
+	"right_amount": (float) - value between -1.0 and 1.0 referring to the wheel 
+position. Positive values are to the right of center,
+	"brake_amount": (float) - value between 0.0 and 1.0 referring to the amount
+				of braking to apply,
+	"drive_mode": (integer) - value indicating the drive mode: -1 = reverse, 
+0 = neutral, 1 = drive	
+}
+```
+
+**response**: (string)
+
+`complete` if the control command was applied, or `no ego vehicle found` if the ego vehicle is not in the simulation.
+
+
+## Weather Configuration
+Configures the weather used in the simulation run. The profiles parameter is optional, and if present, the profiles are added or modified based on the supplied parameters for each profile.
+
+**command ID**:  "WeatherConfig"
+
+**command data**:
+```json
+{
+	"set_profile": (string) - the ID of the weather profile to use,
```

Comment:
Make note that this **must be present in the below configuration**. If this string isn't there, then a random profile will be picked.